### PR TITLE
Add option to set/unset dpi scaling for screens that are not high resolution

### LINKED ISSUE #3489: Display problems on systems that use 'display scaling'
## Description of your problem

**What steps will reproduce the problem?**

1. In the system settings of either Windows 7 or Windows 10, set display scaling to 150%. 
2. Launch Spyder 3.0

**What do you see?**
Many GUI elements look "off", as shown below.

Main screen: the text "Console" in the Help pane has its last character cut off. All text in the main box of the Help pane looks outsized. The icons in the toolbar look outsized with respect to the text labels. 
spyder-3 0_150percent_display_scaling

Preferences screen: the text "WindowsVista" has its last character cut off. Half of the edges of the checkbox icons have been cut off.
spyder-3 0_preferences_150percent_display_scaling

**Please provide any additional information below**
The Spyder GUI looks "proper" again on my system after I manually set the Qt.AA_EnableHighDpiScaling option to False in Spyder's code (in "spyder/app/mainwindow.py"). See below:
spyder-3 0_150percent_display_scaling_qt_option_disabled

spyder-3 0_preferences_150percent_display_scaling_qt_option_disabled

## Versions and main components

* Spyder Version: 3.0.0
* Python Version: 3.5
* Operating system: Windows 7 and Windows 10

## Dependencies

jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  1.5.3 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  1.11.1 (OK)
pandas >=0.13.1  :  0.18.1 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pyflakes >=0.6.0 :  1.3.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
pylint >=0.25    :  1.5.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.3.1 (OK)
sympy >=0.7.3    :  1.0 (OK)

### PR DESCRIPTION
**Pull request by mariacamilaremolinagutierrez**
Fixes #3489

### CODE REVIEW

**Comment by mariacamilaremolinagutierrez** [comment]
This creates a new checkbox in preferences to enable/disable high DPI scaling. The default is disabled. 

pr

**Comment by goanpeca** [comment]
LGTM

File: spyder/app/mainwindow.py [hunk 1]
```diff
@@ -94,11 +94,16 @@
 
 #==============================================================================
 # Proper high DPI scaling is available in Qt >= 5.6.0. This attibute must
-# be set before creating the application
+# be set before creating the application. 
 #==============================================================================
+from spyder.config.main import CONF
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: `CONF` is also imported below (around line `140`). Please remove from there to not import it twice :-)

File: spyder/app/mainwindow.py [hunk 2]
```diff
@@ -94,11 +94,16 @@
 
 #==============================================================================
 # Proper high DPI scaling is available in Qt >= 5.6.0. This attibute must
-# be set before creating the application
+# be set before creating the application. 
 #==============================================================================
+from spyder.config.main import CONF
+if CONF.get('main', 'high_dpi_scaling'):
+    has_high_dpi_scaling=True
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: Please name this variable as `high_dpi_scaling

File: spyder/app/mainwindow.py [hunk 3]
```diff
@@ -94,11 +94,16 @@
...
+from spyder.config.main import CONF
+if CONF.get('main', 'high_dpi_scaling'):
+    has_high_dpi_scaling=True
+else:
+    has_high_dpi_scaling=False
 if hasattr(Qt, 'AA_EnableHighDpiScaling'):
-    QCoreApplication.setAttribute(Qt.AA_EnableHighDpiScaling, True)
-
+    QCoreApplication.setAttribute(Qt.AA_EnableHighDpiScaling, has_high_dpi_scaling)
+
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: Remove this blank addition :-)

**Comment by ccordoba12** [comment]
Great job!! This is exactly what I've done to fix this issue ;-)

I left some minor (and very simple) comments to solve :-)

File: spyder/app/mainwindow.py [hunk 4]
```diff
@@ -261,6 +266,7 @@ def __init__(self, options=None):
 
         qapp = QApplication.instance()
         if PYQT5:
+            # Enabling scpaling for high dpi
```
Comment (goanpeca) [inline spyder/app/mainwindow.py]: Typo scaling vs. scpaling

File: spyder/app/mainwindow.py [hunk 5]
```diff
@@ -261,6 +266,7 @@ def __init__(self, options=None):
 
         qapp = QApplication.instance()
         if PYQT5:
+            # Enabling scpaling for high dpi
             qapp.setAttribute(Qt.AA_UseHighDpiPixmaps)
```
Comment (goanpeca) [inline spyder/app/mainwindow.py]: @ccordoba12 should we also affect this setting with the new option that was added, or it is really not needed?

@SylvainCorlay ?

Comment (ccordoba12) [inline spyder/app/mainwindow.py]: I think the critical one is `HighDpiScaling`. No one has reported problems about this one :-)

Comment (goanpeca) [inline spyder/app/mainwindow.py]: ok

**Comment by goanpeca** [comment]
@mariacamilaremolinagutierrez we normally work on forks of the repo, not on the repo itself.

https://github.com/spyder-ide/spyder/wiki/Dev%3A-Github-Workflow#Developers

**Comment by mariacamilaremolinagutierrez** [comment]
@goanpeca Hi Gonzalo, yes I knew about this after. I apologize, for the next one I'll fork the repo :)

File: spyder/app/mainwindow.py [hunk 6]
```diff
@@ -94,10 +94,15 @@
...
 #==============================================================================
 # Proper high DPI scaling is available in Qt >= 5.6.0. This attibute must
-# be set before creating the application
+# be set before creating the application. 
 #==============================================================================
+from spyder.config.main import CONF
+if CONF.get('main', 'high_dpi_scaling'):
+    high_dpi_scaling=True
+else:
+    high_dpi_scaling=False
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: @mariacamilaremolinagutierrez, please add spaces around equals, like this

    high_dpi_scaling = True

File: spyder/app/mainwindow.py [hunk 7]
```diff
@@ -1171,7 +1177,7 @@ def add_xydoc(text, pathlist):
             for child in self.menuBar().children():
                 if isinstance(child, QMenu) and child != self.help_menu:
                     child.setTearOffEnabled(True)
-
+
```
Comment (ccordoba12) [inline spyder/app/mainwindow.py]: @mariacamilaremolinagutierrez, you forgot to remove this blank :-)

**Comment by ccordoba12** [comment]
@mariacamilaremolinagutierrez, please address my latest comments so I can merge this one :-)

**Comment by mariacamilaremolinagutierrez** [comment]
I don't understand why only that change make a test fail. I'm working on another issue, but as soon as I finish it I'll figure this out. 

image

I don't fully understand the tests it is failing, but they are: 

```
spyder/widgets/sourcecode/tests/test_autocolon.py::test_auto_colon_even_if_colon_inside_quotes xfail
spyder/widgets/sourcecode/tests/test_autocolon.py::test_no_auto_colon_in_listcomp_over_three_lines xfail
spyder/widgets/sourcecode/tests/test_autocolon.py::test_auto_colon_in_two_if_statements_on_one_line xfail
spyder/widgets/sourcecode/tests/test_autoindent.py::test_simple_def xfail
spyder/widgets/sourcecode/tests/test_autoindent.py::test_def_with_unindented_comment xfail
spyder/widgets/sourcecode/tests/test_autoindent.py::test_open_parenthesis xfail
```

**Comment by ccordoba12** [comment]
Thanks @mariacamilaremolinagutierrez! Merging now :-)

**Comment by jitseniesen** [comment]
> I don't fully understand the tests it is failing, but they are: 

@mariacamilaremolinagutierrez In case you haven't figured it out yet: `xfail` indicates that it is a test that we expect to fail, and the test does indeed fail. So it's not a bad thing. We use it when we write tests for bugs that we haven't yet fixed. See http://doc.pytest.org/en/latest/skipping.html for more info.

**Comment by mariacamilaremolinagutierrez** [comment]
Thank you for the explanation @jitseniesen :)

**Pull request merged**